libs/wallet/provider/uphold/uphold_test.go

```diff
@@ -307,7 +307,7 @@ func TestFingerprintCheck(t *testing.T) {
 	var proxy func(*http.Request) (*url.URL, error)
 	wrongFingerprint := "IYSLsapSKlkofKfi6M2hmS4gzXbQKGIX/DHBWIgstw3="
 
-	client = &http.Client{
+	client := &http.Client{
```

Comment:
Note: The behaviour of the test is now different. Previously the test overrode the global `client`.

With the changes in this PR, this no longer the case – this test now operates on its own client, and the rest of code would still be interacting with the global client which has been renamed to `defaultHTTPClient`.

As ugly as this whole story is, I would prefer to keep things consistent, and maybe change this to be as follows, otherwise we would have to refactor quite a lot now (unfeasible) or leave tests in an inconsistent state:

```suggestion
	defaultHTTPClient = &http.Client{
```

Reply:
Regarding the test assigning to the global value. It looks like it was a typo in the test. It does not affect other tests since the test was the last one in the source and none of tests sets `t.Parallel()`. Thus the effect of the that assignment was limited to that test. As the client was deliberately constructed with a wrong fingerprint, if the client will be used in other tests, they will fail to connect.

Reply:
Hopefully it is as you describe, and not actually relying on swapping the global client. Let us see. 🙂